realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -173,6 +160,11 @@ public static void setDefaultSessionErrorHandler(SyncSession.ErrorHandler errorH
      * Gets any cached {@link SyncSession} for the given {@link SyncConfiguration} or create a new one if
      * no one exists.
      *
+     * Note: This will not create a new native (Object Store) session, this will only associate a Realm's path
```

Comment:
This is a public method, so talking about implementation details here is probably not a good idea. It is still relevant info, so perhaps inline inside the method?